pkpdapp/pkpdapp/views/dataset.py

```diff
@@ -0,0 +1,200 @@
+        return context
+
+    def get_paginated_biomarker_dataset(self, context):
+        queryset = Biomarker.objects.filter(
+            biomarker_type__dataset=context['dataset']
+        ).order_by('id')
+        paginator = Paginator(queryset, 20)  # paginate_by
+        page = self.request.GET.get('page')
+        activities = paginator.get_page(page)
+        return activities
+
+
+class DatasetListView(ListView):
+    model = Dataset
+    template_name = 'dataset_list.html'
+
+
+class DatasetCreate(CreateView):
+    model = Dataset
+    fields = ['name', 'description']
+    template_name = 'dataset_form.html'
+
+
+class DatasetUpdate(UpdateView):
+    model = Dataset
+    fields = ['name', 'description']
+    template_name = 'dataset_form.html'
+
+
+def create(request):
+    context = {}
+    if request.method == "POST":
+        form = CreateNewDataset(request.POST, request.FILES)
+        context["form"] = form
+
+        if form.is_valid():
+            uploaded_file = request.FILES['file']
+
+            # handling errors in uploaded files
+
+            # error in file type
+            if not uploaded_file.name.endswith('.csv'):
+                messages.error(request,
+                               BASE_FILE_UPLOAD_ERROR +
+                               'THIS IS NOT A CSV FILE.')
+                return render(request, 'dataset_create.html',
+                              context)
+
+            # error in file content
+            data = pd.read_csv(uploaded_file)
+            colnames = list(data.columns)
+            if len(colnames) > 4:
+                messages.error(
+                    request,
+                    BASE_FILE_UPLOAD_ERROR +
+                    'THIS FILE HAS TOO MANY COLUMNS. ' +
+                    'IT SHOULD ONLY HAVE: subject id, time, biomarker type, ' +
+                    'value')
+                return render(request, 'dataset_create.html',
+                              context)
+            required_cols = ['subject id', 'time', 'biomarker type', 'value']
+            error_cols = []
+            error_string = (BASE_FILE_UPLOAD_ERROR +
+                            'FILE DOES NOT CONTAIN: ')
+            for col in required_cols:
+                if col not in colnames:
+                    error_cols.append(col)
+            if len(error_cols) > 0:
+                messages.error(request,
+                               error_string + ', '.join(error_cols))
+                return render(request, 'dataset_create.html',
+                              context)
+
+            # no errors -> process and save as dataset
+            dataset = Dataset(
+                name=form.cleaned_data["name"],
+                description=form.cleaned_data["description"],
+                datetime=form.cleaned_data["datetime"],
+                administration_type=form.cleaned_data["administration_type"],
+            )
+            dataset.save()
+
+            # add to demo project
```

Comment:
We'll have more than just the demo project. the project that you add the dataset to should come as a parameter in the url, or be set via a field in the form